src/thor/timedistancematrix.cc

```diff
@@ -182,23 +191,24 @@ void TimeDistanceMatrix::Expand(GraphReader& graphreader,
   if (!from_transition && nodeinfo->transition_count() > 0) {
     const NodeTransition* trans = tile->transition(nodeinfo->transition_index());
     for (uint32_t i = 0; i < nodeinfo->transition_count(); ++i, ++trans) {
-      Expand<expansion_direction>(graphreader, trans->endnode(), pred, pred_idx, true);
+      Expand<expansion_direction>(graphreader, trans->endnode(), pred, pred_idx, true, offset_time);
     }
   }
 }
 
 template <const ExpansionType expansion_direction, const bool FORWARD>
 std::vector<TimeDistance> TimeDistanceMatrix::ComputeMatrix(
-    const google::protobuf::RepeatedPtrField<valhalla::Location>& source_location_list,
-    const google::protobuf::RepeatedPtrField<valhalla::Location>& target_location_list,
+    google::protobuf::RepeatedPtrField<valhalla::Location>& source_location_list,
+    google::protobuf::RepeatedPtrField<valhalla::Location>& target_location_list,
     baldr::GraphReader& graphreader,
     const float max_matrix_distance,
     const uint32_t matrix_locations) {
   // Run a series of one to many calls and concatenate the results.
-  const auto& origins = FORWARD ? source_location_list : target_location_list;
-  const auto& destinations = FORWARD ? target_location_list : source_location_list;
+  auto& origins = FORWARD ? source_location_list : target_location_list;
+  auto& destinations = FORWARD ? target_location_list : source_location_list;
```

Comment:
these two lines can be removed. instead `SourceToTarget` can pass in the right order itself